include/onnxruntime/core/framework/data_types_internal.h

```diff
@@ -341,25 +345,51 @@ class MLTypeCallDispatcherRet {
   }
 };
 
-// Version of the MLTypeDispatcher that has an input type which is passed through ('carried')
-// as the first type parameter in the call to Fn when dispatching.
-template <typename TCarried, template <typename, typename> class Fn, typename... Types>
-class MLTypeCallDispatcherWithCarriedType {
+// TODO consolidate this with the other MLTypeCallDispatcher classes
+// can add additional methods to cover their usages, but need to update call sites
+template <typename... Types>
```

Comment:
Comment should explain what the difference is vs MLTypeCallDispatcher (move Fn to Invoke) and why we need this.

May also be work considering replacing MLTypeCallDispatcher with MLTypeCallDispatcher2, but that can be a separate PR. Not sure there's any reason why we need a variant that takes Fn at the class level.

Reply:
i'd like to just have a single dispatcher class if possible. probably in another PR though. in the meantime, i'll add more to this comment

---
In reply to: [564864476](https://github.com/microsoft/onnxruntime/pull/6466#discussion_r564864476) [](ancestors = 564864476)